### PR TITLE
feat(loading): add loading spinner to pages where http request is done

### PR DESCRIPTION
**Pull request by serkan-ozel-2015400123**
New feature:

There is a spinning loading component which is reusable, when htpp requests are made.
Basically it lasts 10 seconds whenever call is completed it is dismissed manually.
 
Thanks

### CODE REVIEW

File: frontend/src/app/auth/providers/auth/auth.service.ts [hunk 1]
```diff
@@ -5,14 +5,20 @@ import { Router } from '@angular/router';
...
 export function tokenGetter() {
   return localStorage.getItem('token');
 }
 
+
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
+  loadingController : LoadingComponent = new LoadingComponent(new LoadingController());
```
Comment (okyksl) [inline frontend/src/app/auth/providers/auth/auth.service.ts]: I don't think we should create components like that. @kemaltulum could you check this out?

Further notes, I don't see where you utilize the `LoadingComponent`, you only use the `LoadingController`. The component itself is unnecessary & redundant.

**Comment by serkan-ozel-2015400123** [comment]
Some changes done including: 
* General module folder is deleted. 
* Using loadingController instead of loadingComponent now. 
* Loading feature does not enter in services now, it is done in ui pages.

File: frontend/src/app/app.module.ts [hunk 2]
```diff
@@ -13,6 +13,8 @@ import { DataModule } from './data/data.module';
 
 import { AppConfigModule } from './app-config.module';
 import { AppRoutingModule } from './app-routing.module';
+import { EventsModule } from "./events/events.module";
+import { GeneralModule } from "./general/general.module";
```
Comment (kemaltulum) [inline frontend/src/app/app.module.ts]: general module gives an error
ERROR in src/app/app.module.ts(17,31): error TS2307: Cannot find module './general/general.module'.
removing general module fixes the error

Comment (serkan-ozel-2015400123) [reply frontend/src/app/app.module.ts]: Thanks I will fix it, or you can too

Comment (serkan-ozel-2015400123) [reply frontend/src/app/app.module.ts]: Done

**Pull request merged**